components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx

```diff
@@ -0,0 +1,130 @@
+                    toast("Your prebuild settings were updated.");
+                },
+            });
+        },
+        [
+            branchMatchingPattern,
+            branchStrategy,
+            configuration.id,
+            configuration.prebuildSettings,
+            interval,
+            toast,
+            updateConfiguration,
+            workspaceClass,
+        ],
+    );
+
+    // TODO: Figure out if there's a better way to deal with grpc enums in the UI
+    const handleBranchStrategyChange = useCallback((val) => {
+        // This is pretty hacky, trying to coerce value into a number and then cast it to the enum type
+        // Would be better if we treated these as strings instead of special enums
+        setBranchStrategy(parseInt(val, 10) as BranchMatchingStrategy);
+    }, []);
+
+    return (
+        <ConfigurationSettingsField>
+            <form onSubmit={handleSubmit}>
+                <Heading3>Prebuild Settings</Heading3>
+                <Subheading className="max-w-lg">These settings will be applied on every Prebuild.</Subheading>
+
+                <InputField
+                    label="Commit interval"
+                    hint="The number of commits to be skipped between prebuild runs."
+                    id="prebuild-interval"
+                >
+                    <input
+                        type="number"
+                        id="prebuild-interval"
+                        min="0"
+                        max="100"
+                        step="5"
+                        value={interval}
+                        onChange={({ target }) => setInterval(target.value)}
+                    />
+                </InputField>
+
+                <SelectInputField
+                    label="Branch Filter"
+                    hint="Run prebuilds on the selected branches only."
+                    value={branchStrategy}
+                    onChange={handleBranchStrategyChange}
+                >
+                    <option value={BranchMatchingStrategy.ALL_BRANCHES}>All branches</option>
+                    <option value={BranchMatchingStrategy.DEFAULT_BRANCH}>Default branch</option>
+                    <option value={BranchMatchingStrategy.MATCHED_BRANCHES}>Match branches by pattern</option>
+                </SelectInputField>
+
+                {branchStrategy === BranchMatchingStrategy.MATCHED_BRANCHES && (
+                    <TextInputField
+                        label="Branch name pattern"
+                        hint="Glob patterns separated by commas are supported."
+                        value={branchMatchingPattern}
+                        onChange={setBranchMatchingPattern}
+                    />
+                )}
+
+                <WorkspaceClassOptions value={workspaceClass} onChange={setWorkspaceClass} />
+
+                <LoadingButton className="mt-8" type="submit" loading={updateConfiguration.isLoading}>
+                    Save
+                </LoadingButton>
```

Comment:
This would add some complexity, but it could be nice to have this disabled when nothing has changed like on the other pages. Up to you though, not sure if it's worth it considering the complexity of the UI itself as well.

Reply:
Yah I decided not too here but am open to doing it in a followup if we think it's worth it. Would want to probably create some abstraction to make it easier though to detect changes.